src/ResourceManager/Sql/Commands.Sql/TransparentDataEncryption/Cmdlet/AddAzureRmSqlManagedInstanceTransparentDataEncryptionCertificate.cs

```diff
@@ -0,0 +1,138 @@
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Commands.Sql.Common;
+using Microsoft.Azure.Commands.Sql.Server.Model;
+using Microsoft.Azure.Commands.Sql.TransparentDataEncryption.Adapter;
+using Microsoft.Azure.Commands.Sql.TransparentDataEncryption.Model;
+using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+using System.Collections.Generic;
+using System.Linq;
+using System.Management.Automation;
+
+
+namespace Microsoft.Azure.Commands.Sql.TransparentDataEncryption.Cmdlet
+{
+    /// <summary>
+    /// Defines the Add-AzureRmSqlServerTransparentDataEncryptionCertificate cmdlet
+    /// </summary>
+    [Cmdlet(VerbsCommon.Add, "AzureRmSqlManagedInstanceTransparentDataEncryptionCertificate", ConfirmImpact = ConfirmImpact.Low, SupportsShouldProcess = true)]
+    [OutputType(typeof(bool))]
+    public class AddAzureRmSqlManagedInstanceTransparentDataEncryptionCertificate : AzureSqlCmdletBase<IEnumerable<AzureRmSqlManagedInstanceTransparentDataEncryptionCertificateModel>, AzureSqlDatabaseTransparentDataEncryptionArmAdapter>
+    {
+        /// <summary>
+        /// Parameter sets
+        /// </summary>
+        protected const string DefaultParameterSet = "AddAzureRmSqlServerTransparentDataEncryptionCertificateDefaultParameterSet";
+
+        /// <summary>
+        /// Gets or sets the resource group name
+        /// </summary>
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = DefaultParameterSet,
+            Position = 0,
+            HelpMessage = "The Resource Group Name")]
+        [ResourceGroupCompleter()]
+        [ValidateNotNullOrEmpty]
+        public override string ResourceGroupName { get; set; }
+
+        /// <summary>
+        /// Gets or sets the agent server name
+        /// </summary>
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = DefaultParameterSet,
+            Position = 1,
+            HelpMessage = "The managed instance name")]
+        [ValidateNotNullOrEmpty]
+        public virtual string ManagedInstanceName { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = DefaultParameterSet,
+            Position = 2,
+            HelpMessage = "The Private blob for Transparent Data Encryption Certificate. For detailed instructions on how to generate the blob go to https://aka.ms/tdecertprep")]
+        [ValidateNotNullOrEmpty]
+        public string PrivateBlob { get; set; }
+
+        /// <summary>
+        /// Gets or sets the Password for Transparent Data Encryption Certificate
+        /// </summary>
+        [Parameter(
+            Mandatory = false,
+            ParameterSetName = DefaultParameterSet,
+            HelpMessage = "The Password for Transparent Data Encryption Certificate")]
+        [ValidateNotNullOrEmpty]
+        public string Password { get; set; }
+
+        /// <summary>
```

Comment:
The cmdlet syntax:
```Powershell
Add-AzureRmSqlManagedInstanceTransparentDataEncryptionCertificate [-ResourceGroupName] <string> [-ManagedInstanceName] <string> [-PrivateBlob] <string> [-Password] <string> [<CommonParameters>]
```
tells us the parameter ```Password``` is required/positional
but in the code it's just optional.